packages/payment-plugin/src/mollie/mollie.handler.ts

```diff
@@ -0,0 +1,115 @@
+import createMollieClient, { RefundStatus } from '@mollie/api-client';
+import { LanguageCode } from '@vendure/common/lib/generated-types';
+import {
+    CreatePaymentResult,
+    CreateRefundResult,
+    Logger,
+    PaymentMethodHandler,
+    PaymentMethodService,
+    SettlePaymentResult,
+} from '@vendure/core';
+
+import { MolliePlugin } from './mollie.plugin';
+
+let paymentMethodService: PaymentMethodService;
+export const molliePaymentHandler = new PaymentMethodHandler({
+    code: 'mollie-payment-handler',
+    description: [
+        {
+            languageCode: LanguageCode.en,
+            value: 'Mollie payment',
+        },
+    ],
+    args: {
+        apiKey: {
+            type: 'string',
+        },
+        redirectUrl: {
+            type: 'string',
+        },
+    },
+    init(injector) {
+        paymentMethodService = injector.get(PaymentMethodService);
+    },
+    createPayment: async (ctx, order, amount, args, _metadata): Promise<CreatePaymentResult> => {
+        try {
+            const { apiKey } = args;
+            let { redirectUrl } = args;
+            redirectUrl = redirectUrl.endsWith('/') ? redirectUrl.slice(0, -1) : redirectUrl; // remove appending slash
+            const paymentMethods = await paymentMethodService.findAll(ctx);
+            const paymentMethod = paymentMethods.items.find(pm =>
```

Comment:
This assumes that there will only be 1 PaymentMethod using that particular mollie api key. Is that a safe assumption? Or could there be a scenario with e.g. 2 PaymentMethods using the same api key but with different redirectUrls? In that case it would be safer to check against both the apiKey and redirectUrl.

Reply:
Fair point. In this case the controller only uses the apiKey from the method, so there is no harm in having the wrong paymentMethod as long as it has the same apiKey.

That said, this is a dangerous assumption for the future, so I fixed it :)